src/Microsoft.DotNet.Tools.Builder/README.md

```diff
@@ -0,0 +1,32 @@
+dotnet-build
+===========
+
+**NAME** 
+dotnet-build -- Orchestrates the compilation of a project and all its dependencies.
+
+**SYNOPSIS**
+dotnet build[options]
+
+**DESCRIPTION**
+
+The build verb orchestrates the compilation of a project: it gathers the dependencies of a project and decides which to compile. 
+
+Users should invoke the Build verb when they want the entire dependency graph compiled, and Compile when they want only a specific project compiled.
+
+Before any compilation begins, the build verb analyzes the project and its dependencies for incremental safety checks. If all checks clear out, then build proceeds with incremental compilation of the project and its dependencies. Otherwise it falls back to non-incremental compilation. Via a profile flag users can choose to receive additional information on how they can improve their build times.
+
+All the projects in the dependency graph that need compilation must pass the following safety checks in order for the compilation process to be incremental:
+- not use pre / post compile scripts
+- not load compilation tools from PATH (e.g., resgen, compilers)
+- use only known compilers (csc, vbc, fsc)
+
```

Comment:
If there is no change to the DLL file i.e., no byte level changes, then the obj produced can still be different in the case of cross module inlining in the JIT. If such an inline-able function changed (ex: a leaf function), then the code in the obj could end up different.

@gkhanna79 or @jkotas, For CPP compilation, what is the granularity?

1. Would we compile: `Multiple IL -> Multiple CPP  -> Single Exe`?
2. Would we compile: `Each IL DLL -> Multiple CPPs -> Multiple Per IL Obj -> Single Exe?
[leaving open the possibility that they could also be IL DLL --> Native DLL with named exports and a thin EXE]

1 would have cross module inlining problem if such an inlinee changed, then it will affect the EXE.
I think our plan is to do 2 which preserves the MSIL DLL byte to OBJ correspondence unless we start doing cross module inlining with CPP also.